### PR TITLE
Reworks Improvised and Overloaded Shotgun Shells.

### PR DESCRIPTION
**Pull request by Impisi**
These things were a little underpowered before( 5 pellets at 3 damage each on improvised, 5 pellets and damage between 1 and 10 on the overloaded). So, I figured I might as well give these guys a rework.

I've taken care to make these things inferior to buckshot shells in every way. Well, except for maybe ascetics. The shower of sparks these things blow across the rooms looks fantastic. Buckshot can reliably deal 30 damage at range and the pellets can travel 50 tiles. Improvised can only travel 8 tiles and reliably deal 10 damage at range. Point blank, buckshot shells will deal 90 damage while improvised shells deal 60 damage. I didn't want some chucklefuck assistant instagibbing people in the halls (well, not without gibbing themselves first...). Range on the pellets is randomized between 1 and 8.

Overloaded shells are more difficult to acquire now as they need liquid plasma in addition to black powder, in addition they fire six pellets and deal three damage each. Now, this shell is going to be a bit more risky to use as the guy firing it tends to get dealt between 30 to 90 brute damage due to how this works. Upon running out of range or hitting something the pellets explode in a moderately sized but weak explosion. Don't worry, its not strong enough to punch through the stations plating but it is enough to send the CE on a killing spree. Anyway, both you and whatever is standing in front of you for 10 range are going to be bathed in explosive glory. A word of warning to any assistant out their with murder on their minds. This thing will knock you out, make a noise loud enough for half the station to hear and if you are brain dead enough to fire it in a maintenance corridor it will puree you into a delicious fuckwit smoothie. Likewise unless you intend to be the hero the station needs firing point blank is also incredibly stupid. So please, murder and/or B&E responsibly!

If my code is absolute shit or you have any balance concerns, please let me know.

:cl: Impisi
   tweak: Improvised shells now deal more damage but are more inaccurate.
   tweak: Overloaded shells now require liquid plasma in addition to black powder. Shells now fire explosive pelets that are good and hurting you and everything in front of you.
   /:cl:

### CODE REVIEW

File: code/modules/projectiles/ammunition/ammo_casings.dm [hunk 1]
```diff
@@ -119,28 +119,23 @@
...
 	materials = list(MAT_METAL=250)
-	pellets = 5
-	variance = 1
-
-/obj/item/ammo_casing/shotgun/improvised/overload/New()
-	..()
-	pellets = rand(3, 8)
-
+	pellets = 6
+	variance = 0.8
```
Comment (Jordie0608) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: This is inherited from the parent now

File: code/modules/projectiles/projectile/bullets.dm [hunk 2]
```diff
@@ -34,16 +34,37 @@
 	damage = 15
 
 /obj/item/projectile/bullet/pellet/weak
+	damage = 6
+	New()
```
Comment (Jordie0608) [inline code/modules/projectiles/projectile/bullets.dm]: We use absolute pathing
`/obj/item/projectile/bullet/pellet/weak/New()`

File: code/modules/projectiles/projectile/bullets.dm [hunk 3]
```diff
@@ -34,16 +34,37 @@
 	damage = 15
 
 /obj/item/projectile/bullet/pellet/weak
+	damage = 6
+	New()
+		range = rand(8)
```
Comment (Jordie0608) [inline code/modules/projectiles/projectile/bullets.dm]: This means random number from 0 to 8, 0 range is probably a bad thing

File: code/modules/projectiles/projectile/bullets.dm [hunk 4]
```diff
@@ -34,16 +34,37 @@
...
+
+/obj/item/projectile/bullet/pellet/weak/on_range()
+	var/datum/effect_system/spark_spread/sparks = new /datum/effect_system/spark_spread
+	sparks.set_up(1, 1, src)
+	sparks.start()
+	..()
+
+/obj/item/projectile/bullet/pellet/overload
 	damage = 3
+	New()
```
Comment (Jordie0608) [inline code/modules/projectiles/projectile/bullets.dm]: See above

File: code/modules/projectiles/projectile/bullets.dm [hunk 5]
```diff
@@ -34,16 +34,37 @@
...
+		range = rand(10)
```
Comment (Jordie0608) [inline code/modules/projectiles/projectile/bullets.dm]: See above

**Comment by Jordie0608** [comment]
Your changelog should actually describe what is being changed

**Comment by Zilenan91** [comment]
Improvised Shells were already inaccurate enough, I can't imagine how inaccurate they are now

**Comment by Impisi** [comment]
Its kind of like grabbing a box of nails and throwing it across the room. Damage should be pretty reliable at 5 range and under though.

Range of zero never seemed to cause problems on testing. Well, it would cause problems on the overcharged shell but that's OK. Blowing yourself up is a design feature.

I'll get to work on the rest. I'm sure I can monkey this stuff so that it meets coding standards and works at the same time

File: code/modules/projectiles/projectile/bullets.dm [hunk 6]
```diff
@@ -34,16 +34,37 @@
...
 
-/obj/item/projectile/bullet/pellet/random/New()
-	damage = rand(10)
+/obj/item/projectile/bullet/pellet/overload/on_hit(atom/target, blocked = 0)
+	..()
+	explosion(target, 0, 0, 2)
```
Comment (KazeEspada) [inline code/modules/projectiles/projectile/bullets.dm]: This is a lot more powerful then you think. My minigun could level hallways with -1,-1,1.

Comment (Impisi) [inline code/modules/projectiles/projectile/bullets.dm]: Level as in punch holes through the tiles and damage wiring? Your issue might be the two negative values. I found that my problem with hullbreaches went away when I set my devastation and high damage values to 0.

File: code/modules/projectiles/ammunition/ammo_casings.dm (hunk 1 above)
Comment (Impisi) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: Lol brb sepuku.

File: code/modules/projectiles/projectile/bullets.dm (hunk 6 above)
Comment (KazeEspada) [inline code/modules/projectiles/projectile/bullets.dm]: I meant like break every window and wall that the bullet hit. Never any floor damage.

Comment (Impisi) [inline code/modules/projectiles/projectile/bullets.dm]: Oh yeah. It wrecks windows. Walls are somewhat tough but multiple rounds should mess up regular walls.

Using it usually knocks you down for about twenty seconds. I'm alright with walls and windows getting pulverised due to the effective counter play.

**Comment by Impisi** [comment]
Well I managed to fix everything else but now the damn overloaded shells spawn as already fired.  ¯\_(ツ)_/¯

**Comment by Impisi** [comment]
Fuck, my changes are on a different branch and I can't merge. I'll make a new PR

**Pull request closed without merging**